kavanabhat/Spark-Benchmarks-Setup
# New changes

Readme:
Correct the location of logs:

install.sh

1) The script assumes it will be invoked from the spark-perf-setup directory which is incorrect
TPCDS benchmark relies on WORKDIR. You can expect the user to set it to where the Spark-Benchmarks-Setup is cloned.
So, PERFUTILS_DIR will be $WORKDIR/Spark-Benchmarks-Setup/spark-perf-setup. Check if WORKDIR is set at the start of the script.

2) HDFS URL should be set as 
# URL of the HDFS installation in the Spark EC2 cluster
HDFS_URL = "hdfs://sparkhab12:9000/test/"

It should be like http://sparkhab12.aus.stglabs.ibm.com:50070

3) Following variables should be assigned boolean values instead of string. This is causing spark cluster to be stopped and restarted
# Whether to restart the Master and all Workers
# This should always be false for Yarn
RESTART_SPARK_CLUSTER = "False"
RESTART_SPARK_CLUSTER = "False"

# Rsync SPARK_HOME to all the slaves or not
RSYNC_SPARK_HOME = "False"
 
4) Set these to true at install time and set to false in runbench.sh after execution of the tests.

# Which tests to prepare. Set this to true for the first
# installation or whenever you make a change to the tests.
PREP_SPARK_TESTS = True
PREP_PYSPARK_TESTS = False
PREP_STREAMING_TESTS = False
PREP_MLLIB_TESTS = False

5) Create a Spark-Benchmarks-Setup/spark-perf-setup/wdir and under it create spark-perf and spark_perf_logs. Add "spark-perf-setup/wdir" to .gitignore.

runbench.sh

1) PERFUTILS_DIR should be based on $WORKDIR.  Check if WORKDIR is set at the start of the script.

2) Set PREP_SPARK_TESTS, PREP_PYSPARK_TESTS, PREP_STREAMING_TESTS, PREP_MLLIB_TESTS as mentioned in #5 above

3) Display the default scaling factor in the following prompt

Do you wish to change scaling factor? [y/N] y

4) The script gets stuck waiting for input from the user as shown below. So, set PROMPT_FOR_DELETES to false in config.py

About to remove all files and directories under /home/testuser/spark-2.0.1-bin-hadoop2.7/work on ['localhost'], is this ok? [y, n]

Further validating whether data is getting generated in the HDFS location.
